### PR TITLE
[ci][nightly][core] Fix many nodes test due to default list of state APIs

### LINKED ISSUE #31011: [core][nightly] many_nodes test failed with dashboard test
### What happened + What you expected to happen

https://buildkite.com/ray-project/release-tests-branch/builds/1248#0184f908-b524-429f-8017-2f3fa2d33e2c

```
Waiting for memory monitor actor to be ready...
Memory monitor actor is ready now.
Traceback (most recent call last):
  File "distributed/test_many_tasks.py", line 139, in <module>
    test()
  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "distributed/test_many_tasks.py", line 80, in test
    dashboard_test = DashboardTestAtScale(addr)
  File "/home/ray/oss-release-tests-2/distributed/dashboard_test.py", line 89, in __init__
    node = list(filter(lambda n: n["node_ip"] == current_node_ip, nodes))[0]
IndexError: list index out of range
```

### Versions / Dependencies

master

### Reproduction script

NA

### Issue Severity

None

### PR DESCRIPTION
**Pull request by rickyyx**
Signed-off-by: rickyyx <rickyx@anyscale.com>

## Why are these changes needed?

many_nodes failure: https://github.com/ray-project/ray/issues/31011

## Related issue number
Closes https://github.com/ray-project/ray/issues/31011

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rickyyx** [comment]
Runnign many_nodes, many_actors, many_tasks here: https://buildkite.com/ray-project/release-tests-pr/builds/23374

File: release/benchmarks/distributed/dashboard_test.py [hunk 1]
```diff
@@ -85,8 +85,11 @@ def __init__(self, addr: ray._private.worker.RayContext):
 
         # Schedule the actor on the current node (which is a head node).
         current_node_ip = ray._private.worker.global_worker.node_ip_address
-        nodes = ray.experimental.state.api.list_nodes()
-        node = list(filter(lambda n: n["node_ip"] == current_node_ip, nodes))[0]
+        nodes = ray.experimental.state.api.list_nodes(
+            filters=[("node_ip", "=", current_node_ip)]
```
Comment (rkooo567) [inline release/benchmarks/distributed/dashboard_test.py:89]: Setting `limit=10000`?

**Review by rkooo567 — COMMENTED** [review]
Thanks for the fix!

File: release/benchmarks/distributed/dashboard_test.py (hunk 1 above)
Comment (rkooo567) [reply release/benchmarks/distributed/dashboard_test.py:89]: or actually this should also fix the issue...

**Review by rkooo567 — APPROVED** [review]

**Comment by rkooo567** [comment]
Looks like the previous run failed at "wait_clusters". Rerunnig

**Comment by rkooo567** [comment]
Rerun failed by the same reason. Let's just merge it and see how it goes...

**Pull request merged**